src/main/webapp/app/components/calendarButton/UsageAnalysisCalendarButton.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import {
+  InputGroup,
+  Button,
+  Dropdown,
+  DropdownButton,
+  Row,
+  Form,
+  Col,
+  Container,
+} from 'react-bootstrap';
+import classnames from 'classnames';
+import { getMomentInstance } from 'app/shared/utils/Utils';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import moment from 'moment';
+import {
+  APP_LOCAL_DATETIME_FORMAT_Z_FORCE,
+  APP_LOCAL_DATE_FORMAT,
+} from 'app/config/constants';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+import pluralize from 'pluralize';
+
+type CalendarButtonProps = {
+  currentDate: string;
+  currentFromDate: string;
+  currentToDate: string;
+  fromDate: (newDate: string) => void;
+  toDate: (newDate: string) => void;
+  filterToggled: (filterActive: boolean) => void;
+};
+
+export const UsageAnalysisCalendarButton: React.FunctionComponent<CalendarButtonProps> = props => {
+  function handleFromDayChange(day: any) {
```

Comment:
We should really try to avoid using `any`. It defeats the purpose of using TS. Could you take a look and update if possible? This includes the type for `filters` below.